frontend/src/components/Discover_Event/filter-section/FilterSection.tsx

```diff
@@ -20,21 +49,46 @@ const FilterSection = () => {
       ) {
         setShowCalendar(false);
       }
+      if (
+        catergoryRef.current &&
```

Comment:
This has a weird interaction, for instance if we click on the categories again it keep rerender the page while the dropdown remain open whereas it should not do anything, and for the radio click if you select it and select it again it should toggle off the selection same goes for pick a date, if you keep clicking on the pick a date radio button, it keep opening and not close. Can do some investigate to solve  this problem, using conditonal is a good approach like line 77 with &&, perhaps considering not using Ref.

Reply:
1. Change from `Ref` to just conditional for `Category`, `Pick a date`, and `Event Type` (No data structure to hold selected categories and event types yet
2. Add Scroll Area for long sections like `Category` and `Event Type

https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/2c49eadf-3802-4e5c-9357-8c246b5103d6


https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/d0fecce1-889a-4d7f-b0d5-40db16977217


https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/d1139e98-e2fa-4d50-8b92-7a975891969b